core/src/main/java/com/datastax/oss/driver/api/core/Cluster.java

```diff
@@ -33,9 +33,10 @@
  *     CqlSession}.
  */
 public interface Cluster<SessionT extends Session> extends AsyncAutoCloseable {
+
   /** Returns a builder to create a new instance of the default implementation. */
-  static ClusterBuilder builder() {
-    return new ClusterBuilder();
+  static DefaultClusterBuilder builder() {
```

Comment:
We could return the generic type here, but it's a bit intimidating with the self type. So I chose to return the implementation (which means it has to be in the public API, but that's not a problem since it contains almost nothing).